runtime/src/lib.rs

```diff
@@ -535,24 +535,23 @@ impl_runtime_apis! {
 		}
 
 		// Estimate gas for execute script.
-		fn estimate_gas_execute_script(account: AccountId, transaction: Vec<u8>, cheque_limit: u128) -> Result<MoveApiEstimation, DispatchError> {
+		fn estimate_gas_execute_script(_account: AccountId, transaction_bc: Vec<u8>, _cheque_limit: u128) -> Result<MoveApiEstimation, DispatchError> {
 			// Main input for the VM are these script parameters.
 			let pallet_move::ScriptTransaction {
 				bytecode,
 				args,
 				type_args,
-			} = pallet_move::ScriptTransaction::try_from(transaction.as_ref())
+			} = pallet_move::ScriptTransaction::try_from(transaction_bc.as_ref())
 				.map_err(|_| pallet_move::Error::<Runtime>::InvalidScriptTransaction)?;
 			let args: Vec<&[u8]> = args.iter().map(AsRef::as_ref).collect();
 
 			// Make sure the script parameters are valid.
-			pallet_move::verify_script_integrity_and_check_signers(&bytecode).map_err(pallet_move::Error::<Runtime>::from)?;
+			let signer_count = pallet_move::verify_script_integrity_and_check_signers(&bytecode).map_err(pallet_move::Error::<Runtime>::from)?;
 
 			// `limited_balance` cannot perform any actual updates to balances.
-			let mut limited_balance = pallet_move::balance::BalanceAdapter::<Runtime>::new();
-			limited_balance.write_cheque(&account, &cheque_limit)?;
+			let unlimited_balance = pallet_move::balance::BalanceAdapter::<Runtime>::for_dry_run(&args, signer_count)?;
```

Comment:
The comment above should be updated with the variable name, or it can be removed.

Maybe `mocked_balance` (or something) would be a better name. The `limited_balance` implied it's limited in the sense it cannot actually update the balance 😃

Reply:
Updated :+1: